### ISSUE TITLE
Wrong reference to code line

### ISSUE THREAD

**Issue by erez-o** [doxygen/doxygen#7461]
Hi, 

I'm seeing multiple occurrences of references from the documentation to code lines that are incorrect. 

For example, `Definition at line 879 of file amqp_framing.h.` while the code line is actually 813.

I'm attaching a zip file of the project, the doxygen.cfg and the output html.

[rabbitmq-c-master.zip](https://github.com/doxygen/doxygen/files/4000259/rabbitmq-c-master.zip)
[doxygen.txt](https://github.com/doxygen/doxygen/files/4000264/doxygen.txt)
[rabbitmq3_html_output.zip](https://github.com/doxygen/doxygen/files/4000265/rabbitmq3_html_output.zip)

Inside the html output, you can look at file `structamqp__access__properties__t.html`, or any of the structures following it in the sidebar.

Note: I did some trial and error and saw that when ENABLE_PREPROCESSING=NO, the code lines are correct.

Edited: The doxygen version I'm using is not the latest, it's from 30.3.2019, see https://github.com/doxygen/doxygen/tree/16d025c8a08b485f5d43ade0986d716bd6aa06f8

Thanks

**Comment by albert-github** [comment]
One thing that is a bit strange is the line:
```
PREDEFINED             = AMQP_BEGIN_DECLS= \ AMQP_END_DECLS= \ AMQP_PUBLIC_FUNCTION= \ AMQP_PUBLIC_VARIABLE= \ AMQP_CALL= \ AMQP_DEPRECATED(x)=x
```
I think this should be:
```
PREDEFINED             = AMQP_BEGIN_DECLS= \
                         AMQP_END_DECLS= \
                         AMQP_PUBLIC_FUNCTION= \
                         AMQP_PUBLIC_VARIABLE= \
                         AMQP_CALL= \
                         AMQP_DEPRECATED(x)=x
```
though this is not causing the mentioned problems.

(tested against the current master)

**Comment by albert-github** [comment]
I've just pushed a proposed patch, pull request #7462

**Comment by erez-o** [comment]
Thanks for the speedy response. I'll test it when it's merged.

Regarding your comment about `PREDEFINED` - will doxygen give a different output the way I wrote it? is the newline needed or just for readability?

**Comment by albert-github** [comment]
Code has been integrated in master on github (please don't close the issue as this will be done at the moment of a release).

**Comment by albert-github** [comment]
When using a `PREDEFINED` like:
```
PREDEFINED = A= \ B=
```
We see when running the preprocessor (`-d preprocessor` e,g. on an empty `aa.h`):
```
Preprocessor output (size: 0 bytes):
---------
00001
---------
Macros accessible in this file:
---------
A B \
---------
```

So doxygen thinks there is also a `macro` with the name `\`. Either use:
```
PREDEFINED = A= \
             B=
```
or
```
PREDEFINED = A=  B=
```

**Comment by erez-o** [comment]
Thanks for the clarification.

I briefly tested and saw that you require `\` followed by a `\n`, so for example, `\` followed by space then `\n` will also create an unnecessary macro.

If I may suggest, I think it's best if the [docs](http://www.doxygen.nl/manual/preprocessing.html) are edited to explicitly state that separation is done by a space or `\` immediately followed by a new-line character, and also print a warning if you found a macro with the name `\` (I can't think of any reason to have such a macro).

**Comment by albert-github** [comment]
In the documentation of the configuration settings (http://www.doxygen.nl/manual/config.html) it is already stated:

> Multiple lines can be concatenated by inserting a backslash (\) as the last character of a line.

**Comment by erez-o** [comment]
Thx, I missed that.

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.8.17.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
